02-peer-protocol.md

```diff
@@ -986,6 +994,8 @@ A receiving node:
   - if other `id` violations occur:
     - MAY send a `warning` and close the connection, or send an
       `error` and fail the channel.
+  - if `blinding_point` is provided:
+    - MUST use the corresponding blinded private key to decrypt the `onion_routing_packet` (see [Route Blinding](04-onion-routing.md#route-blinding))
```

Comment:
The blinded key is used for both the `onion_routing_packet` and the `encrypted_recipient_data` inside of it. The onion can't be encrypted with the public key of the recipient because the sender doesn't know it, the sender only knows the blinded public key, so everywhere the recipient would normally use it's private key, it needs to use the blinded version instead.

Reply:
The trick is that when you're part of a blinded route, the onion isn't encrypted for your `node_id`, it's encrypted to a blinded version of your `node_id`, so you must derive the corresponding private key to be able to decrypt the onion.

Let's detail a sample scenario, with a route `Alice -> Bob -> Carol`. With a normal payment, the onion is created using `nodeId(Alice)`, `nodeId(Bob)` and `nodeId(Carol)` and each node can use their corresponding private key to do a DH and decrypt the onion.

If we use route blinding, we'll have something like `Alice -> Blinded(Bob) -> Blinded(Carol)`. The onion doesn't use `nodeId(Bob)` and `nodeId(Carol)` anymore, but different public keys that have been derived from `nodeId(Bob)` and `nodeId(Carol)`. When Bob receives the message, he must derive the corresponding private key to be able to decrypt the onion. He can derive that blinded private key  based on his normal private key and the `blinding_point` provided in a tlv in `update_add_htlc`.

Reply:
oh yes of course! Makes sense. Thanks!